metadata/install-qa-check.d/60python-site

```diff
@@ -0,0 +1,229 @@
+# Copyright 2019-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# QA checks related to site-packages directory:
+# - missing, mismatched or stray .pyc files
+# Maintainer: Python project <python@gentoo.org>
+
+python_site_check() {
+	local save=$(shopt -p nullglob)
+	shopt -s nullglob
+	local progs=( "${EPREFIX}"/usr/lib/python-exec/*/gpep517 )
+	local bad_libdirs=()
+	[[ $(get_libdir) != lib ]] && bad_libdirs=(
+		"${ED}/usr/$(get_libdir)"/{python3,pypy}*
+	)
+	${save}
+
+	local forbidden_package_names=(
+		# NB: setuptools/discovery.py is a good source of ideas
+		benchmark benchmarks dist doc docs examples scripts tasks
+		test tests tools util utils
+		# catch double-prefix installs, e.g. https://bugs.gentoo.org/618134
+		lib $(get_libdir) usr
+		.pytest_cache .hypothesis _trial_temp
+	)
+
+	local invalid=()
+	local mismatched_timestamp=()
+	local mismatched_data=()
+	local missing=()
+	local stray=()
+
+	local bad_versions=()
+	local outside_site=()
+	local stray_packages=()
+
+	# Avoid running the check if sufficiently new gpep517 is not installed
+	# yet. It's valid to schedule (for merge order) >=gpep517-8 after
+	# packages which have this check run if they don't use distutils-r1.
+	if [[ ${EAPI} == [0123] ]] || ! nonfatal has_version ">=dev-python/gpep517-8" ; then
+		return
+	fi
+
+	for prog in "${progs[@]}"; do
+		local impl=${prog%/*}
+		impl=${impl##*/}
+
+		# NB: using ${impl}* to catch pypy3.* for pypy3
+		local pydir=( "${ED}"/usr/lib/${impl}* )
+		[[ -d ${pydir} ]] || continue
+
+		# check for packages installing outside site-packages
+		case ${CATEGORY}/${PN} in
+			dev-lang/python|dev-python/pypy*)
+				;;
+			*)
+				mapfile -d $'\0' -O "${#outside_site[@]}" -t outside_site < <(
+					find "${pydir}" -mindepth 1 -maxdepth 1 \
+						'!' -name site-packages -print0
+				)
+				;;
+		esac
+
+		local sitedir=( "${pydir}"/site-packages )
+		[[ -d ${sitedir} ]] || continue
+
+		# check for bad package names or versions
+		mapfile -d $'\0' -O "${#bad_versions[@]}" -t bad_versions < <(
+			find "${sitedir}" -maxdepth 1 '(' \
+				-name '*-0.0.0.dist-info' -o \
+				-name '*UNKNOWN*.dist-info' -o \
+				-name '*-0.0.0.egg-info' -o \
+				-name '*UNKNOWN*.egg-info' \
+				')' -print0
+		)
+
+		# check for stray files in site-packages
+		mapfile -d$'\0' -t -O "${#stray_packages[@]}" stray_packages < <(
```

Comment:
missing space `-d $'\0'`, this makes it misbehave for some reason

On a side-note, you can use `mapfile -d ''` rather than `$'\0'`

Reply:
Sigh. I really do wish this comment would've happened before I rewrote everything to use `while`-`read` loop.

Reply:
> missing space `-d $'\0'`, this makes it misbehave for some reason

Well, reason-wise I think it doesn't see it and then took `-` as the delimiter, `-t` was lost so you end up with entries terminated with `-`. Not sure what happens with the `t` in this context though... feel like mapfile should've error'ed out. Ah well.